plugins/org.jkiss.dbeaver.ext.datavirtuality.ui/src/org/jkiss/dbeaver/ext/datavirtuality/views/DataVirtualityConnectionPage.java

```diff
@@ -0,0 +1,290 @@
+            testLink.addSelectionListener(new SelectionAdapter() {
+                @Override
+                public void widgetSelected(SelectionEvent e) {
+                    site.testConnection();
+                }
+            });
+        }
+
+        {
+            Composite addrGroup = UIUtils.createControlGroup(control, DataVirtualityMessages.label_security, 4, 0, 0);
+            addrGroup.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+
+            UIUtils.createControlLabel(addrGroup, DataVirtualityMessages.label_user);
+
+            usernameText = new Text(addrGroup, SWT.BORDER);
+            GridData gd = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING | GridData.FILL_HORIZONTAL);
+            usernameText.setLayoutData(gd);
+            usernameText.addModifyListener(textListener);
+
+            UIUtils.createEmptyLabel(addrGroup, 2, 1);
+
+            Text passwordText = createPasswordText(addrGroup, DataVirtualityMessages.label_password);
+            gd = new GridData(GridData.HORIZONTAL_ALIGN_BEGINNING | GridData.FILL_HORIZONTAL);
+            passwordText.setLayoutData(gd);
+            passwordText.addModifyListener(textListener);
+
+            createPasswordControls(addrGroup, 2);
+        }
+
+        createDriverPanel(control);
+        setControl(control);
+    }
+
+    @Override
+    public boolean isComplete()
+    {
+        return hostText != null &&
+            !CommonUtils.isEmpty(hostText.getText());
+    }
+
+    @Override
+    public void loadSettings()
+    {
+        super.loadSettings();
+
+        // Load values from new connection info
+        DBPConnectionConfiguration connectionInfo = site.getActiveDataSource().getConnectionConfiguration();
+        if (hostText != null) {
+            if (CommonUtils.isEmpty(connectionInfo.getHostName())) {
+                hostText.setText(DataVirtualityConstants.DEFAULT_HOST_PREFIX);
+            } else {
+                hostText.setText(connectionInfo.getHostName());
+            }
+        }
+        if (portText != null) {
+            if (!CommonUtils.isEmpty(connectionInfo.getHostPort())) {
+                portText.setText(String.valueOf(connectionInfo.getHostPort()));
+            } else if (site.getDriver().getDefaultPort() != null) {
+                portText.setText(site.getDriver().getDefaultPort());
+            } else {
+                portText.setText(""); //$NON-NLS-1$
+            }
+        }
+        if (dbText != null) {
+            String databaseName = connectionInfo.getDatabaseName();
+            if (CommonUtils.isEmpty(databaseName)) {
+                databaseName = DataVirtualityConstants.DEFAULT_DB_NAME;
+            }
+            dbText.setText(databaseName);
+        }
+        // TODO: need to fix for checkbox
+        if (sslCheckbox != null) {
+            sslCheckbox.setSelection(CommonUtils.notEmpty(connectionInfo.getProviderProperty(DataVirtualityConstants.PROP_SSL))=="mms" ? true : false);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.